documentation/config_logging_advanced.md

```diff
@@ -0,0 +1,124 @@
+
+#### Adding a JSON version of the main Puppet Server logs
+
+To configure Puppet Server to log its main logs to a second log file, in JSON format, simply add a section like this to your `logback.xml` file (it should be added at the same level in the XML as your existing appender/appenders, but the order of the appenders does not matter):
+
+``` xml
+<appender name="JSON" class="ch.qos.logback.core.rolling.RollingFileAppender">
+        <file>/var/log/puppetlabs/puppetserver/puppetserver.log.json</file>
+
+        <rollingPolicy class="ch.qos.logback.core.rolling.TimeBasedRollingPolicy">
+            <fileNamePattern>/var/log/puppetlabs/puppetserver/puppetserver.log.json.%d{yyyy-MM-dd}</fileNamePattern>
+            <maxHistory>5</maxHistory>
+        </rollingPolicy>
+
+        <encoder class="net.logstash.logback.encoder.LogstashEncoder"/>
+    </appender>
+```
+
+Logback writes logs using components called [appenders](http://logback.qos.ch/manual/appenders.html).  Note that the above configuration includes the `RollingFileAppender`, to rotate the log files and ensure they don't fill up your disk.  Also, note that there are a lot of configuration options available for the `LogstashEncoder` above, including the ability to modify the list of fields that you wish to include, or give them different field names.  For more info see the [Logstash Logback Encoder Docs](https://github.com/logstash/logstash-logback-encoder/blob/master/README.md#loggingevent-fields).
+
+The above XML snippet will create an appender that will log to the specified JSON file when active.  In order to activate the appender, you also need to add an `appender-ref`, e.g. in the `<root>` section of your `logback.xml`:
+
+``` xml
+<root level="info">
+        <appender-ref ref="FILE"/>
+        <appender-ref ref="JSON"/>
+</root>
+```
+
+You could also comment out or remove the other `appender-ref`s, if you decided you *only* wanted to log the JSON format.
+
+#### Adding a JSON version of the Puppet Server HTTP Access logs
+
+To add JSON logging for Puppet Server's HTTP requests, modify the `request-logging.xml` file.  Here's an example of adding a logback appender there:
+
+``` xml
+   <appender name="JSON" class="ch.qos.logback.core.rolling.RollingFileAppender">
+        <file>/var/log/puppetlabs/puppetserver/puppetserver-access.log.json</file>
+
+        <rollingPolicy class="ch.qos.logback.core.rolling.TimeBasedRollingPolicy">
+            <fileNamePattern>/var/log/puppetlabs/puppetserver/puppetserver-access.log.json.%d{yyyy-MM-dd}</fileNamePattern>
+            <maxHistory>30</maxHistory>
+        </rollingPolicy>
+
+      <encoder class="net.logstash.logback.encoder.AccessEventCompositeJsonEncoder">
+          <providers>
+              <version/>
+              <pattern>
+                  <pattern>
+                    {
+                      "@timestamp":"%date{yyyy-MM-dd'T'HH:mm:ss.SSSXXX}",
```

Comment:
This field is generated by Logstash, but I wouldn't say it's required or standard. Newer versions of Kibana let you choose a time field. If you were interested in knowing when Logstash processed the event and when it was generated you would want two different time fields. However, it's really up to the user. I think if they're already Logstash users they'll understand this either way.